packages/core-flows/src/price-list/workflows/batch-price-list-prices.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  BatchPriceListPricesWorkflowDTO,
+  BatchPriceListPricesWorkflowResult,
+} from "@medusajs/types"
+import {
+  WorkflowData,
+  createWorkflow,
+  parallelize,
+  transform,
+} from "@medusajs/workflows-sdk"
+import {
+  createPriceListPricesStep,
+  removePriceListPricesStep,
+  updatePriceListPricesStep,
+  validatePriceListsStep,
+  validateVariantPriceLinksStep,
+} from "../steps"
+
+export const batchPriceListPricesWorkflowId = "batch-price-list-prices"
+export const batchPriceListPricesWorkflow = createWorkflow(
+  batchPriceListPricesWorkflowId,
+  (
+    input: WorkflowData<{
+      data: BatchPriceListPricesWorkflowDTO
+    }>
+  ): WorkflowData<BatchPriceListPricesWorkflowResult> => {
+    const [_, variantPriceCreateMap, variantPriceUpdateMap] = parallelize(
+      validatePriceListsStep([{ id: input.data.id }]),
+      validateVariantPriceLinksStep([{ prices: input.data.create }]),
+      validateVariantPriceLinksStep([{ prices: input.data.update }]).config({
+        name: "variant-price-link-map-update",
+      })
+    )
+
+    const createPriceListPricesInput = transform(
+      { data: input.data, map: variantPriceCreateMap },
+      ({ data, map }) => ({
+        data: [{ id: data.id, prices: data.create }],
+        variant_price_map: map,
+      })
+    )
+
+    const updatePriceListPricesInput = transform(
+      { data: input.data, map: variantPriceUpdateMap },
+      ({ data, map }) => ({
+        data: [{ id: data.id, prices: data.update }],
+        variant_price_map: map,
+      })
+    )
+
+    const [created, updated, deleted] = parallelize(
```

Comment:
I think it's better to have a `batch-price-list-prices` step, and inside it call the `*-price-list-prices` workflows instead, since there might be more to a workflow of creating a price than just calling this one step, so it's more future-proof like that.

Check the product workflows to see how I've done it after some discussions with Adrien and Carlos.